src/server.cpp

```diff
@@ -2760,16 +2665,35 @@ void Server::sendDetachedInventory(const std::string &name, session_t peer_id)
 
 void Server::sendDetachedInventories(session_t peer_id, bool incremental)
 {
-	for (const auto &detached_inventory : m_detached_inventories) {
-		const std::string &name = detached_inventory.first;
+	// Lookup player name, to filter detached inventories just after
+	std::string peer_name;
+	if (peer_id != PEER_ID_INEXISTENT) {
+		RemotePlayer *player = m_env->getPlayer(peer_id);
+		if (player) {
+			peer_name = player->getName();
+		}
+	}
+
+	auto send_cb = [this, peer_id, peer_name, incremental](ServerInventoryMgr *imgr,
+			const std::string &name, Inventory *inv) {
 		if (incremental) {
-			Inventory *inv = detached_inventory.second;
 			if (!inv || !inv->checkModified())
-				continue;
+				return;
 		}
 
-		sendDetachedInventory(name, peer_id);
-	}
+		// if we are pushing inventories to a specific player
+		// we should filter to send only the right inventories
+		if (!peer_name.empty()) {
+			std::string attached_player;
+			imgr->getAttachedPlayer(name, attached_player);
```

Comment:
Following would be more common and easier to understand:
`std::string attached_player = imgr->getAttachedPlayer(name);`

Reply:
obviously yes